include/kernel.h

```diff
@@ -3925,6 +3942,168 @@ extern void _timer_expiration_handler(struct _timeout *t);
 
 #endif /* _ARCH_DECLARE_STACK */
 
+/**
+ * @defgroup mem_domain_apis Memory domain APIs
```

Comment:
Need to add a section in doc/api/kernel_api.rst for this Memory domain APIs defgroup